### ISSUE TITLE
Harfbuzz package fails to set the required variables used by the project

### ISSUE THREAD

**Issue by netcorefan1** [microsoft/vcpkg#31152]
**Describe the bug**
The only variables set by vcpkg toolchain are HARFBUZZ_DIR, HARFBUZZ_LIBRARY_DEBUG, HARFBUZZ_LIBRARY_RELEASE while I believe the project also expects HARFBUZZ_LIBRARIES.

**Environment**
- OS: Windows 11
- Compiler: latest
- harfbuzz 7.1.0
- freetype  2.12.1#3

**To Reproduce**
It's a bit involved task. One should clone [emgucv](https://github.com/emgucv/emgucv) and use the vcpkg toolchain to link against the installed packages rather than directly build from source. Hope the below failure log alone would be enough to identify the source of the issue.

**Expected behavior**
The expected variables correctly populated rather than being forced to manually set them.

**Failure logs**
```
-- Setting up FREETYPE
-- Found ZLIB: optimized;C:/dev/vcpkg/installed/x64-windows/lib/zlib.lib;debug;C:/dev/vcpkg/installed/x64-windows/debug/lib/zlibd.lib (found version "1.2.13")
-- FREETYPE:   C:/dev/vcpkg/installed/x64-windows/share/freetype (ver 2.12.1)
-- FREETYPE_LIBRARIES: C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll
CMake Warning (dev) in C:/dev/vcpkg/installed/x64-windows/share/harfbuzz/harfbuzzConfig.cmake:
  Policy CMP0011 is not set: Included scripts do automatic cmake_policy PUSH
  and POP.  Run "cmake --help-policy CMP0011" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  The included script

    C:/dev/vcpkg/installed/x64-windows/share/harfbuzz/harfbuzzConfig.cmake

  affects policy settings.  CMake is implying the NO_POLICY_SCOPE option for
  compatibility, so the effects are applied to the including context.
Call Stack (most recent call first):
  C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:852 (_find_package)
  CMakeLists.txt:132 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- HARFBUZZ:    C:/dev/vcpkg/installed/x64-windows/share/harfbuzz (ver )
-- HARFBUZZ_LIBRARIES: HARFBUZZ_LIBRARY-NOTFOUND
```

```
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
HARFBUZZ_LIBRARY
    linked by target "opencv_freetype" in directory C:/Emugcv47_V2/emgucv/opencv_contrib/modules/freetype
    linked by target "cvextern" in directory C:/Emugcv47_V2/emgucv/Emgu.CV.Extern

CMake Warning (dev) at Emgu.CV.Extern/CMakeLists.txt:3825 (ADD_DEPENDENCIES):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "zlib" of target "cvextern" does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:632 (_add_library):
  Policy CMP0003 should be set before this line.  Add code such as

    if(COMMAND cmake_policy)
      cmake_policy(SET CMP0003 NEW)
    endif(COMMAND cmake_policy)

  as early as possible but after the most recent call to
  cmake_minimum_required or cmake_policy(VERSION).  This warning appears
  because target "opencv_freetype" links to some libraries for which the
  linker must search:

    HARFBUZZ_LIBRARY-NOTFOUND

  and other libraries with known full path:

    C:/Emugcv47_V2/emgucv/build_x86_64/bin/runtimes/win-x64/native/Debug/opencv_imgproc470d.lib
    C:/Emugcv47_V2/emgucv/build_x86_64/opencv/3rdparty/lib/Debug/ippiwd.lib
    C:/Emugcv47_V2/emgucv/build_x86_64/opencv/3rdparty/ippicv/ippicv_win/icv/lib/intel64/ippicvmt.lib
    C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll

  CMake is adding directories in the second list to the linker search path in
  case they are needed to find libraries from the first list (for backwards
  compatibility with CMake 2.4).  Set policy CMP0003 to OLD or NEW to enable
  or disable this behavior explicitly.  Run "cmake --help-policy CMP0003" for
  more information.
Call Stack (most recent call first):
  opencv/cmake/OpenCVUtils.cmake:1560 (add_library)
  opencv/cmake/OpenCVModule.cmake:993 (ocv_add_library)
  opencv/cmake/OpenCVModule.cmake:905 (_ocv_create_module)
  opencv/cmake/OpenCVModule.cmake:1139 (ocv_create_module)
  opencv_contrib/modules/freetype/CMakeLists.txt:24 (ocv_define_module)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by dg0yt** [comment]
> Policy CMP0011 is not set

The CMake project you are building is not setting `cmake_minimum_required(VERSION ...)` to a reasonable value, *before*  calling `project(...)`. This causes many sub-issues.

**Comment by SchaichAlonso** [comment]
`HARFBUZZ_LIBRARY`, `HARFBUZZ_INCLUDE_DIR` and so on are not needed to be set manually. Instead, the `HARFBUZZ_ROOT` variable can be provided to point to the directory containting the `harfbuzzConfig.cmake` file, in case you want to prefer a provided harfbuzz over the one in a default search path.

The vcpkg toolchain will, however, add the vcpkg paths to the default search paths, therefore HARFBUZZ_ROOT can be omitted, as you either want to use vcpkg's harfbuzz, or you don't add harfbuzz to the [manifest](https://learn.microsoft.com/en-us/vcpkg/users/manifests) ([emgucv](https://github.com/emgucv/emgucv) doesn't use manifest mode it seems).

I would, however, disencourage the mixture of vcpkg dependencies and operating-system provided dependencies. Instead, you should either provide a Dockerfile or similar functionality with the operating system's package repository locked in the dockerfile and _all_ dependencies installed by the OS (potentially needs you to build your own packages), or to submodule vcpkg and have vcpkg handle _all_ dependencies.... for the sake of tracability. [Classic mode](https://learn.microsoft.com/en-us/vcpkg/users/classic-mode) needs vcpkg to be downloaded by a developer and manually bootstrapped and installed, and all developers are lazy and will just go 
```
git clone [vcpkg ](https://github.com/microsoft/vcpkg.git)
cd vcpkg
./bootstrap.bat
vcpkg install $my_dependency
``` 
... which results on each checkout taking vcpkg's master branch and each developer having a different version of the vcpkg provided dependencies. Submoduling vcpkg results on the vcpkg version - and changes thereof - to the explicit in the consuming project's repostiory, so a change in vcpkg's ports that causes a breakage of the consuming project appears as a commit in the consumer's history, and if the breakage is not immediately detected, can be bisected by checking out previous versions of the consuming repostiory rather then having to bisect through combinations of vcpkg and the consumer (and potentially the OS package repository on top of both).

emgucv uses a historic version of harfbuzz . If you need to use an old harfbuzz (API changes or whatnot) in an otherwise new vcpkg deployment, you can use something like
```
   "dependencies": [
      "icu",
      "harfbuzz",
      "pango"
    ],
    "builtin-baseline": "501db0f17ef6df184fcdbfbe0f87cde2313b6ab1",
    "overrides": [
      {
        "name": "harfbuzz",
        "version": "4.2.0#1"
      }
    ]
```
To have vcpkg deploy `icu` and `pango` in the versions that are in (2023.04.15)[https://github.com/microsoft/vcpkg/releases/tag/2023.04.15] which is (501db0f17ef6df184fcdbfbe0f87cde2313b6ab1)[https://github.com/microsoft/vcpkg/commit/501db0f17ef6df184fcdbfbe0f87cde2313b6ab1], but use version `4.2.0` revision `1` for harfbuzz instead of 2023.04.15's default of `7.1.0`. `harfbuzz-4.1.0#1` will also be used when building `pango`. Note that the new pango might fail to build with the old harfbuzz, in which case you need to override the pango to a previous version, too.

The versions available are documented in the `versions` subdirectory of vcpkg, e.g. [versions/h-/harfbuzz.json](https://github.com/microsoft/vcpkg/blob/2023.04.15/versions/h-/harfbuzz.json) for harfbuzz.

**Comment by netcorefan1** [comment]
> > Policy CMP0011 is not set
> 
> The CMake project you are building is not setting `cmake_minimum_required(VERSION ...)` to a reasonable value, _before_ calling `project(...)`. This causes many sub-issues.

You're a genius! Without your comment I would have never been able to fix this. Initially, changing from 3.16 to 3.25 didn't made any difference. Once I placed `CMAKE_MINIMUM_REQUIRED` before `PROJECT(emgucv)` all warnings disappeared.
Should I send a pull request for that [file](https://github.com/emgucv/emgucv/blob/master/CMakeLists.txt) to ask the author to change from:
```
PROJECT(emgucv)
CMAKE_MINIMUM_REQUIRED(VERSION 3.16)
```
into:
```
CMAKE_MINIMUM_REQUIRED(VERSION 3.25)
PROJECT(emgucv)
```
???

Unfortunately the configuration still fails:
```
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
HARFBUZZ_LIBRARY
    linked by target "opencv_freetype" in directory C:/Emugcv47_V2/emgucv/opencv_contrib/modules/freetype
    linked by target "cvextern" in directory C:/Emugcv47_V2/emgucv/Emgu.CV.Extern

CMake Error at Emgu.CV.Extern/CMakeLists.txt:3825 (ADD_DEPENDENCIES):
  The dependency target "zlib" of target "cvextern" does not exist.


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```
At least now this only happens at the end and the entire log appears clean from warning or issues of any kind. It was really hard to understand something before this fix!

**Comment by dg0yt** [comment]
These are problems external to vcpkg. It is up to you to talk to them.

There is no standard (legacy) Find module  for Harfbuzz. So many projects use there own vendored find modules (e.g. Find Harfbuzz, and you have to look at these module and maybe provide input variables. Or patch. Harfbuzz provides CMake config now, but not every project will already make use of it.

> The only variables set by vcpkg toolchain are HARFBUZZ_DIR, HARFBUZZ_LIBRARY_DEBUG, HARFBUZZ_LIBRARY_RELEASE while I believe the project also expects HARFBUZZ_LIBRARIES.

Usually `<Pkg>_LIBRARIES` is derived from `<Pkg>_LIBRARY_DEBUG` and `<Pkg>_LIBRARY_RELEASE`, 
https://cmake.org/cmake/help/latest/module/SelectLibraryConfigurations.html.

**Comment by netcorefan1** [comment]
@SchaichAlonso I updated the log after the change suggested by @dg0yt . So, I'm not sure on what part of your comment I should focus on.

> `HARFBUZZ_LIBRARY`, `HARFBUZZ_INCLUDE_DIR` and so on are not needed to be set manually. Instead, the `HARFBUZZ_ROOT` variable can be provided to point to the directory containting the `harfbuzzConfig.cmake` file, in case you want to prefer a provided harfbuzz over the one in a default search path.

Emgucv compiles its own [submodule](https://github.com/harfbuzz/harfbuzz/tree/aee123fc83388b8f5acfb301d87bd92eccc5b843) during the build process. It is version 4.3.0. This, and pretty much all the other libraries are very old and this is the main reason for which I'm trying to link vcpkg ones.
This is the section that build freetype, after some platform compatibility checks:
```
:BUILD_FREETYPE
cd 3rdParty
cd freetype2
IF NOT EXIST %BUILD_FOLDER% mkdir %BUILD_FOLDER%
cd %BUILD_FOLDER%
%CMAKE% -G %CMAKE_CONF% %GENERAL_CMAKE_CONFIG_FLAGS% ^
-DCMAKE_DISABLE_FIND_PACKAGE_ZLIB:BOOL=TRUE ^
-DCMAKE_DISABLE_FIND_PACKAGE_BZip2:BOOL=TRUE ^
-DCMAKE_DISABLE_FIND_PACKAGE_PNG:BOOL=TRUE ^
-DCMAKE_DISABLE_FIND_PACKAGE_HarfBuzz:BOOL=TRUE ^
..
%CMAKE% --build . --config Release --target INSTALL
cd ..
cd ..
cd ..
SET FREETYPE_DIR=%cd%\3rdParty\freetype2\%BUILD_FOLDER%
REM SET EMGU_CV_CMAKE_CONFIG_FLAGS=%EMGU_CV_CMAKE_CONFIG_FLAGS% -DFreetype_DIR:STRING=%FREETYPE_DIR% 

:BUILD_HARBUZZ
cd harfbuzz
IF NOT EXIST %BUILD_FOLDER% mkdir %BUILD_FOLDER%
cd %BUILD_FOLDER%
%CMAKE% -G %CMAKE_CONF% %GENERAL_CMAKE_CONFIG_FLAGS% -DHB_HAVE_FREETYPE:BOOL=TRUE ..
%CMAKE% --build . --config Release --target INSTALL
cd ..
cd ..
REM SET HARFBUZZ_DIR=%cd%\harfbuzz\%BUILD_FOLDER%
REM SET EMGU_CV_CMAKE_CONFIG_FLAGS=%EMGU_CV_CMAKE_CONFIG_FLAGS% -DHarfbuzz_DIR:STRING=%HARFBUZZ_DIR%
```
I skipped this and added `-DBUILD_ZLIB:BOOL=FALSE` to avoid it being compiled from source.
It is correctly recognized:
```
-- General configuration for OpenCV 4.7.0
-- ZLib: optimized C:/dev/vcpkg/installed/x64-windows/lib/zlib.lib debug C:/dev/vcpkg/installed/x64-windows/debug/lib/zlibd.lib (ver 1.2.13)
```
But I still get:
```
HARFBUZZ_LIBRARY
    linked by target "opencv_freetype" in directory C:/Emugcv47_V2/emgucv/opencv_contrib/modules/freetype
    linked by target "cvextern" in directory C:/Emugcv47_V2/emgucv/Emgu.CV.Extern

CMake Error at Emgu.CV.Extern/CMakeLists.txt:3825 (ADD_DEPENDENCIES):
  The dependency target "zlib" of target "cvextern" does not exist.
```
With `-DBUILD_ZLIB:BOOL=TRUE` the zlib error disappears and remains the HARFBUZZ_LIBRARY not found error. I disabled BUILD_ZLIB because I could imagine how things could become messed up with two different zlib sources (vcpkg and emgucv compiled one).

It is fine for me to use the one installed by default from vcpkg, therefore I believe that HARFBUZZ_ROOT can be omitted.

Emgucv don't use any vcpkg manifest or, to say better, don't use vcpkg at all. Some libraries are compiled as cloned submodules, other are just dist downloaded and compiled.
Initially I used vcpkg.json manifest, but I switched to classic mode. A simple `vcpkg update and install package1 package2 etc` within the install script seemed  much cleaner. In the remote case that a particular version of a package breaks things, I can always switch back to manifest. I'm also using overlay ports, so keeping things shared at global level can greatly simplify development.

I have no experience at all with Dockerfile, but it seems to me like having a huge separate environment. Something like running a WSL separately just to run some stuff. I prefer to do everything natively within Windows. All the builds are for my personal use, so I'm perfectly fine even in classic mode.

However, if I have understood well, you're confirming what has always been my main concern. Perhaps I have been too naive when I thought that I could just replace everything with vcpkg packages, link the toolchain, adjust some configuration stuff and done.  There is much more under the hood that can make the transition pure nightmare. Some emgucv libraries can contains custom commits and I'm afraid that this alone is enough to break everything and make any fix very hard. Emgucv is a wrapper, if for any reason a signature changes I will get into runtime issues which are pure hell because they could happens at any time.
The manifest trick you showed can help to mitigate the risks, but I feel like I'll have to lose a good amount of time in monitoring changes and other related tasks.
I'm wondering if I should abort this and let emgucv (and other projects like this) compile the stuff for which has been designed even if old or at least just use vcpkg for libraries that could be considered safe under this point of view and freetype-harfbuzz does not seems to belong to such category.
Initially (after some adjustments) I successfully got it working with hdf5, tiff, libgeotiff, proj. As soon as I decided to include freetype and harfbuzz to the transition things became very complicated.
Are you also trying in emgucv (or similar projects) to make such kind of transitions? Wrappers like these are very enjoyable because it allows to develop everything in C#, but it is very annoying that all of them suffer from update issues.

**Comment by netcorefan1** [comment]
> These are problems external to vcpkg. It is up to you to talk to them.

I'll send them a pull request as soon as possible. This bug cause serious breakages with vcpkg.

> There is no standard (legacy) Find module for Harfbuzz. So many projects use there own vendored find modules (e.g. Find Harfbuzz, and you have to look at these module and maybe provide input variables. Or patch. Harfbuzz provides CMake config now, but not every project will already make use of it.
> 
> > The only variables set by vcpkg toolchain are HARFBUZZ_DIR, HARFBUZZ_LIBRARY_DEBUG, HARFBUZZ_LIBRARY_RELEASE while I believe the project also expects HARFBUZZ_LIBRARIES.
> 
> Usually `<Pkg>_LIBRARIES` is derived from `<Pkg>_LIBRARY_DEBUG` and `<Pkg>_LIBRARY_RELEASE`, https://cmake.org/cmake/help/latest/module/SelectLibraryConfigurations.html.

Thanks for the detailed explanation. I want to avoid any "How this? How that?". I'm going to study the link you posted along with your precious tips. I'll post my findings as soon as I can get back with something concrete.

**Comment by netcorefan1** [comment]
@dg0yt Your comment has been precious and allowed me to discover the sources of all the issues although I still have doubts on how would be the cleanest solution to applies fixes without resorting to what looks like dirty patches.
This:
```
find_package(HARFBUZZ CONFIG) 
if(NOT HARFBUZZ_FOUND)
	message(STATUS "HARFBUZZ:    NO")
else()
	message(STATUS "HARFBUZZ:    ${HARFBUZZ_DIR} (ver ${HARFBUZZ_VERSION})")
	SET(HARFBUZZ_INCLUDE_DIRS "${HARFBUZZ_DIR}/../../../include/harfbuzz")
	get_target_property(HARFBUZZ_LIBRARY harfbuzz::harfbuzz IMPORTED_LOCATION_RELEASE)
	SET(HARFBUZZ_LIBRARIES ${HARFBUZZ_LIBRARY})
	SET(HARFBUZZ_LINK_LIBRARIES ${HARFBUZZ_LIBRARY})
	SET(HARFBUZZ_LINK_LIBRARIES_XXXXX ${HARFBUZZ_LINK_LIBRARIES})
```
results in:
```
-- HARFBUZZ:    C:/dev/vcpkg/installed/x64-windows/share/harfbuzz (ver )
-- HARFBUZZ_LIBRARIES: HARFBUZZ_LIBRARY-NOTFOUND
```
It fails here: `get_target_property(HARFBUZZ_LIBRARY harfbuzz::harfbuzz IMPORTED_LOCATION_RELEASE)`.
Unlike freetype which calls `set_target_properties` to set such properties, `..\share\harfbuzz\harfbuzzConfig.cmake` seems to set this variable to an empty value:
`set(HARFBUZZ_LIBRARY harfbuzz::harfbuzz CACHE INTERNAL "")`
If I have understood correctly, today I learned the concept of legacy modules detection you previosly mentioned: freetype is using legacy method while harfbuzz the new method (with no legacy support) and the project is expecting harfbuzz legacy. Not sure why HARFBUZZ_VERSION is empty.

Unfortunately, "-DHARFBUZZ_LIBRARY=vcpkgpath" does not work. The above code override it with an empty value. The only think I can think of is patching:
`set(HARFBUZZ_LIBRARY ${HARFBUZZ_LIBRARY_RELEASE})`

Regarding zlib, `findpackage(zlib)` is not called on Windows:
```
IF(WIN32 OR APPLE OR NETFX_CORE OR IS_UBUNTU)
  #always build from source on windows / Mac
  MESSAGE(STATUS "Building zlib from source")
  ADD_DEPENDENCIES(${the_target} zlib)
  SET(ZLIB_INCLUDE_DIR "${OPENCV_SUBFOLDER}/3rdparty/zlib")
  SET(ZLIB_LIBRARY "zlib")
  INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
ELSE()
  FIND_PACKAGE(ZLIB)
  IF(ZLIB_FOUND)
    MESSAGE(STATUS "CVEXTERN: ZLIB found.")
  ELSE()
    MESSAGE(STATUS "CVEXTERN: ZLIB not found, building from source")
    ADD_DEPENDENCIES(${the_target} zlib)
    SET(ZLIB_INCLUDE_DIR "${OPENCV_SUBFOLDER}/3rdparty/zlib")
    SET(ZLIB_LIBRARY "zlib")
  ENDIF()
  INCLUDE_DIRECTORIES(${ZLIB_INCLUDE_DIR})
ENDIF()
```
Even here, the only thing I can think of is a dirty patch (or, should I say hack?):
```
# IF(WIN32 OR APPLE OR NETFX_CORE OR IS_UBUNTU)
IF(APPLE OR NETFX_CORE OR IS_UBUNTU)
```
Well, after all I'm satisfied. All issues has been identified along with their fixes. They only seems to me very dirty and hope you can have ideas better than mines.

**Comment by dg0yt** [comment]
> get_target_property(HARFBUZZ_LIBRARY harfbuzz::harfbuzz IMPORTED_LOCATION_RELEASE)

I tried to understand why this leads to "HARFBUZZ_LIBRARY-NOTFOUND": Harfbuzz doesn't really come with official "exported" CMake config, but this file is added by the vcpkg port and operates like an old-school find module. It relies on internal vcpkg variables  for locating the actual library. So you must use the vcpkg toolchain for this to work.

**Comment by netcorefan1** [comment]
> > get_target_property(HARFBUZZ_LIBRARY harfbuzz::harfbuzz IMPORTED_LOCATION_RELEASE)
> 
> I tried to understand why this leads to "HARFBUZZ_LIBRARY-NOTFOUND": Harfbuzz doesn't really come with official "exported" CMake config, but this file is added by the vcpkg port and operates like an old-school find module. It relies on internal vcpkg variables for locating the actual library. So you must use the vcpkg toolchain for this to work.

I'm afraid I spoke too soon!

You means `<VcpkgEnabled>true</VcpkgEnabled>` on the generated project, right?
If find myself in the following situations when building:
- With vcpkg ENABLED and HARFBUZZ_LIBRARY-NOTFOUND (no change to source) - ALL FREETYPE STUFF FOUND
- With vcpkg DISABLED (DEFAULT) and HARFBUZZ_LIBRARY (fixed in source through `set(HARFBUZZ_LIBRARY ${HARFBUZZ_LIBRARY_RELEASE})`) - ALL FREETYPE STUFF FOUND
- With vcpkg DISABLED (DEFAULT) and HARFBUZZ_LIBRARY-NOTFOUND (no change to source) - Unable to find FREETYPE header ('ft2build.h')

In all the 3 cases above, compilations fails when it comes the time to link harfbuzz (`Cannot open include file: 'hb.h': No such file or directory`) and I'm surprised because all the needed info are there:
`CMakeVars.txt`:
```
CMAKE_TOOLCHAIN_FILE=C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake
CMAKE_TRY_COMPILE_CONFIGURATION=Release
CMAKE_TRY_COMPILE_PLATFORM_VARIABLES=VCPKG_TARGET_TRIPLET;VCPKG_TARGET_ARCHITECTURE;VCPKG_APPLOCAL_DEPS;VCPKG_CHAINLOAD_TOOLCHAIN_FILE;Z_VCPKG_ROOT_DIR
FREETYPE_CONFIG=C:/dev/vcpkg/installed/x64-windows/share/freetype/freetype-config.cmake
FREETYPE_CONSIDERED_CONFIGS=C:/dev/vcpkg/installed/x64-windows/share/freetype/freetype-config.cmake
FREETYPE_CONSIDERED_VERSIONS=2.12.1
FREETYPE_DIR=C:/dev/vcpkg/installed/x64-windows/share/freetype
FREETYPE_FOUND=1
FREETYPE_INCLUDE_DIRS=C:/dev/vcpkg/installed/x64-windows/share/freetype/../../../include/freetype2
FREETYPE_LIBRARIES=C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll
FREETYPE_LIBRARIES=C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll
FREETYPE_LIBRARY=freetype
FREETYPE_LINK_LIBRARIES=C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll
FREETYPE_LINK_LIBRARIES_XXXXX=C:/dev/vcpkg/installed/x64-windows/bin/freetype.dll
FREETYPE_VERSION=2.12.1
FREETYPE_VERSION_COUNT=3
FREETYPE_VERSION_MAJOR=2
FREETYPE_VERSION_MINOR=12
FREETYPE_VERSION_PATCH=1
FREETYPE_VERSION_TWEAK=0
HARFBUZZ_CONFIG=C:/dev/vcpkg/installed/x64-windows/share/harfbuzz/HARFBUZZConfig.cmake
HARFBUZZ_CONSIDERED_CONFIGS=C:/dev/vcpkg/installed/x64-windows/share/harfbuzz/HARFBUZZConfig.cmake
HARFBUZZ_CONSIDERED_VERSIONS=unknown
HARFBUZZ_DIR=C:/dev/vcpkg/installed/x64-windows/share/harfbuzz
HARFBUZZ_FEATURES=core
HARFBUZZ_FOUND=1
HARFBUZZ_INCLUDE_DIR=C:/dev/vcpkg/installed/x64-windows/include/harfbuzz
HARFBUZZ_INCLUDE_DIRS=C:/dev/vcpkg/installed/x64-windows/share/harfbuzz/../../../include/harfbuzz
HARFBUZZ_LIBRARIES=HARFBUZZ_LIBRARY-NOTFOUND
HARFBUZZ_LIBRARY=HARFBUZZ_LIBRARY-NOTFOUND
HARFBUZZ_LIBRARY=HARFBUZZ_LIBRARY-NOTFOUND
HARFBUZZ_LIBRARY_DEBUG=C:/dev/vcpkg/installed/x64-windows/debug/lib/harfbuzz.lib
HARFBUZZ_LIBRARY_RELEASE=C:/dev/vcpkg/installed/x64-windows/lib/harfbuzz.lib
HARFBUZZ_LINK_LIBRARIES=HARFBUZZ_LIBRARY-NOTFOUND
HARFBUZZ_LINK_LIBRARIES_XXXXX=HARFBUZZ_LIBRARY-NOTFOUND
HARFBUZZ_VERSION_COUNT=0
HARFBUZZ_VERSION_MAJOR=0
HARFBUZZ_VERSION_MINOR=0
HARFBUZZ_VERSION_PATCH=0
HARFBUZZ_VERSION_TWEAK=0
```
I really don't know if there is something wrong or not in the Harfbuzz port, but as you said, the toolchain should be able to find the HB headers just like does for FT ones.

In addition to that, when trying to manually set headers I also got a `.../freetype.dll : fatal error LNK1107: invalid or corrupt file: cannot read at 0x308`. I'm afraid this has something to do with the [concerns](https://github.com/microsoft/vcpkg/issues/31152#issuecomment-1528052711) I expressed to @SchaichAlonso although I cannot be sure until I manage to fix the issues above.

**Comment by netcorefan1** [comment]
I want to thanks @dg0yt . I finally managed to get this working and his last comment has been precious. I spent hours in trying to set variables here and there and without any result until I have done what should have been done from the beginning.
I replaced completely the freetype and harfbuzz section with:
```
find_package(FREETYPE CONFIG)
find_package(HARFBUZZ CONFIG REQUIRED)
```
And then replaced:
`target_link_libraries(${the_target} ${FREETYPE_LIBRARY} ${HARFBUZZ_LIBRARY})`
With:
`set_target_properties(opencv_freetype PROPERTIES VS_GLOBAL_VcpkgEnabled TRUE)`

Three lines of code and the project compiles well using 100% vcpkg natively and without any mess between debug and release mode.

I also want to thanks @SchaichAlonso for his precious suggestions on using manifests. Now manifest mode has become a must for development!
